packages/contract-schema/spec/abi.spec.json

```diff
@@ -144,37 +144,75 @@
 
     "Parameter": {
       "type": "object",
-      "properties": {
-        "name": { "$ref": "#/definitions/Name" },
-        "type": { "$ref": "#/definitions/Type" },
-        "components": {
-          "type": "array",
-          "items": { "type": "object" }
-        },
-	"internalType": { "type": "string" }
-      },
       "if": {
         "properties": {
           "type": {
             "oneOf": [
-              { "pattern": "^tuple$" }
+              { "pattern": "^tuple" }
             ]
           }
         }
       },
-      "then": { "required": ["name", "type", "components"] },
-      "else": { "required": ["name", "type"] }
+      "then": {
+        "properties": {
+          "name": { "$ref": "#/definitions/Name" },
+          "type": { "$ref": "#/definitions/Type" },
+          "components": {
+            "type": "array",
+            "items": { "$ref": "#/definitions/Parameter" },
+            "minItems": 1
```

Comment:
For compatibility with Solidity 0.4.x, I think you should remove this requirement; Solidity 0.4.x allows empty structs.